src/odb_pack.c

```diff
@@ -367,12 +370,28 @@ static int pack_backend__freshen(
 	git_odb_backend *backend, const git_oid *oid)
 {
 	struct git_pack_entry e;
+	struct p_timeval times[2];
+	time_t now;
 	int error;
 
 	if ((error = pack_entry_find(&e, (struct pack_backend *)backend, oid)) < 0)
 		return error;
 
-	return git_futils_touch(e.p->pack_name);
+	now = time(NULL);
+
+	if (e.p->last_freshen > now - FRESHEN_FREQUENCY)
+		return 0;
+
+	times[0].tv_sec = now;
+	times[0].tv_usec = 0;
+	times[1].tv_sec = now;
+	times[1].tv_usec = 0;
+
+	if ((error = p_utimes(e.p->pack_name, times)) < 0)
+		return git_path_set_error(errno, e.p->pack_name, "touch");
```

Comment:
Why is the `git_futils_touch` call expanded? Ah, now I see, probably to avoid the second `time(NULL)` call? Maybe include `time_t` in `git_futils_touch(file)`, instead of duplicating logic? If it is set to `NULL`, `git_futils_touch` may still compute its own `time_t`.